src/ValueParsers/DecimalParser.php

```diff
@@ -66,9 +65,9 @@ private function getMath() {
 	 *         does not use scientific notation, $exponent will be 0.
 	 */
 	public function splitDecimalExponent( $valueString ) {
-		if ( preg_match( '/^(.*)([eE]|x10\^)([-+]?[,\d]+)$/', $valueString, $matches ) ) {
-			$exponent = $this->normalizeDecimal( $matches[3] );
-			return array( $matches[1], intval( $exponent ) );
+		if ( preg_match( '/^(.*)(?:[eE]|x10\^)([-+]?[\d,]+)$/', $valueString, $matches ) ) {
```

Comment:
The only reason I can think of is that @brightbyte wanted to make this parser a little bit more flexible so it accepts exponents with group separators. This is tested, see `12e3,4`.